sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -490,65 +495,74 @@ export class ApplicationService extends RecordDataModelService<Application> {
       })
       .getOne();
   }
+
   /**
-   * Fetches application by applicationId and userId (optional).
-   * @param applicationId
-   * @param userId
-   * @returns
+   * Gets a student application by applicationId.
+   * Student id can be provided for authorization purposes.
+   * @param applicationId application id.
+   * @param studentId optional student id for authorization purposes.
+   * @returns student application.
    */
-  async getApplicationByIdAndUser(
+  async getApplicationById(
     applicationId: number,
-    userId?: number,
+    options?: { loadDynamicData?: boolean; studentId?: number },
   ): Promise<Application> {
-    const applicationQuery = this.repo
-      .createQueryBuilder("application")
-      .select([
-        "application.id",
-        "application.data",
-        "application.applicationStatus",
-        "application.pirStatus",
-        "application.applicationStatusUpdatedOn",
-        "application.applicationNumber",
-        "application.pirDeniedOtherDesc",
-        "currentAssessment.id",
-        "currentAssessment.assessmentWorkflowId",
-        "currentAssessment.noaApprovalStatus",
-        "offering.id",
-        "offering.offeringIntensity",
-        "offering.studyStartDate",
-        "offering.studyEndDate",
-        "offering.offeringStatus",
-        "location.id",
-        "location.name",
-        "pirDeniedReasonId.id",
-        "pirDeniedReasonId.reason",
-        "programYear.id",
-        "programYear.formName",
-        "programYear.startDate",
-        "programYear.endDate",
-        "applicationException.exceptionStatus",
-        "application.submittedDate",
-      ])
-      .leftJoin("application.currentAssessment", "currentAssessment")
-      .leftJoin("currentAssessment.offering", "offering")
-      .leftJoin("application.location", "location")
-      .leftJoin("location.institution", "institution")
-      .leftJoin("institution.institutionType", "institutionType")
-      .innerJoin("application.student", "student")
-      .innerJoin("student.user", "user")
-      .innerJoin("application.programYear", "programYear")
-      .leftJoin("application.pirDeniedReasonId", "pirDeniedReasonId")
-      .leftJoin("application.applicationException", "applicationException")
-      .where("application.id = :applicationIdParam", {
-        applicationIdParam: applicationId,
-      })
-      .andWhere("application.applicationStatus != :overwrittenStatus", {
-        overwrittenStatus: ApplicationStatus.overwritten,
-      });
-    if (userId) {
-      applicationQuery.andWhere("user.id = :userId", { userId });
-    }
-    return applicationQuery.getOne();
+    return this.repo.findOne({
+      select: {
+        id: true,
+        data: options?.loadDynamicData ? (true as unknown) : undefined,
```

Comment:
can't we use `false`?
`  data: (options?.loadDynamicData ? true : false) as unknown,`

Reply:
Changed.